design-proposals/vpro-rm.md

```diff
@@ -254,6 +254,74 @@ Edge Infrastructure Manager team will implement the following functionality to s
 - Intelligement Remote Power Management
 - Device de-activation
 
+### Implementation plan for Admin Control Mode
+
+To enable Admin Control Mode, the DM Manager implementation will be updated to activate devices using Admin Control
+Mode (ACM) or Client Control Mode (CCM). The proposed plan for the release is:
+
+- Extend inventory to store the DNS Suffix and activation mode (ACM or CCM) alongside the instance ID.
+  - DNS Suffix added from UI or CLI during set up.
+- Extend DM Manager device activation to check for activation mode from inventory.
+- Add domain profile creation when activation mode is set to admin.
+- DM Manager retrieves DNS Suffix from inventory.
+- DM Manager retrieves provisioning certificate and password from Vault for domain profile creation.
+- Update AMT profile configuration to align with the requested activation mode selected.
+
+Additional updates to support admin mode are documented in the [OpenDMT documentation](./vpro-opendmt.md).
+
+> Note: this will be included in the next release after the 2025.2 release.
+
+Device onboarding and activation will be performed in the following flow when running in admin mode.
+
+```mermaid
+sequenceDiagram
+  title: AMT device provisioning and activation
+  %%{wrap}%%
+  autonumber
+  participant US as User
+  participant UI as UI/CLI
+  participant ON as Onboarding
+  participant DM as DM Manager
+  participant VT as Vault
+  participant IN as Inventory
+  participant PM as Platform Manageability Agent (PMA)
+  participant RP as RPS
+  activate UI
+  US ->> UI: Set activation mode as admin
+  US ->> UI: Set DNS Suffix
+  US ->> UI: Upload Provisioning Certificate
+  US ->> UI: Upload Provisioning Certificate password
+  UI ->> VT: Store Provisioning Certificate
+  UI ->> VT: Store Provisioning Certificate password
+  UI ->> IN: Store DNS Suffix for edge node instance
```

Comment:
It has to be single inventory API call to store both DNS suffix and activation mode in host instance resource.